fixing issue Orphan NativePeer method with Java 11.0.20.1 #421

adysinghh opened the pull request:
fix #421

cyrille-artho commented on src/peers/gov/nasa/jpf/vm/JPF_jdk_internal_misc_Unsafe.java:
```diff
@@ -43,10 +45,10 @@
     int objRef = env.getStaticReferenceField("jdk.internal.misc.Unsafe", "theUnsafe");
     return objRef;
-  }
+  } 
```
Please try to avoid introducing extra formatting changes, as this may make it more difficult to merge your patch in other branches later.

cyrille-artho commented on src/peers/gov/nasa/jpf/vm/JPF_jdk_internal_misc_Unsafe.java:
```diff
@@ -60,7 +62,7 @@ public void registerNatives____V(MJIEnv env, int clsObjRef) {}
       return 0;
   }
-
+  
```
Try to remove this change as well.

cyrille-artho replied:
Can you please revert the second change that swaps the two methods, so we have a patch that only has the first (functional) change?

cyrille-artho commented on src/peers/gov/nasa/jpf/vm/JPF_jdk_internal_misc_Unsafe.java:
```diff
@@ -26,8 +26,10 @@
 import gov.nasa.jpf.vm.NativePeer;
 import gov.nasa.jpf.vm.SystemState;
 import gov.nasa.jpf.vm.ThreadInfo;
+import junit.Test;
```
Are these two imports needed? You do not (yet) add a new test in this patch.

cyrille-artho reviewed (changes requested):
Thank you for the update. Please try to minimize the changes; if it is not possible to add a unit test to check against warnings of JPF, then there is no need to add imports for the `junit` test packages.

adysinghh commented:
@cyrille-artho Hi, I had incorporated all the suggested changes

cyrille-artho commented on src/peers/gov/nasa/jpf/vm/JPF_jdk_internal_misc_Unsafe.java:
```diff
@@ -43,24 +44,21 @@
-  public void registerNatives____V(MJIEnv env, int clsObjRef) {}
-
-  @MJI
-  public int addressSize0____I (MJIEnv env, int objRef) {
```
This function does not seem to be tested by the current unit tests, but are you sure it is not needed at all?

cyrille-artho commented on src/peers/gov/nasa/jpf/vm/JPF_jdk_internal_misc_Unsafe.java:
```diff
@@ -43,24 +44,21 @@
 
   @MJI
-  public long objectFieldOffset__Ljava_lang_reflect_Field_2__J (MJIEnv env, int unsafeRef, int fieldRef) {
+  public long objectFieldOffset__Ljava_lang_reflect_Field_2__J(MJIEnv env, int unsafeRef, int fieldRef) {
```
Please add the space before the "(" back so we do not have extra changes in the patch; this is important to make future merges to other branches easier.

cyrille-artho reviewed (changes requested):
Thank you for the update. This version is a step forward but still has some changes that do not seem to relate directly to the incorrectly named native peer.
Can you please try to make a minimal patch that does not change any other lines? There are several changes (I've highlighted some of them) that add or remove spaces or empty lines. There is also one change that removes an @MJI method. While the tests still all pass, there may be functionality that the unit tests do not cover but rely on that @MJI method.

adysinghh commented:
@cyrille-artho Thanks for the valuable feedback! I've incorporated the suggested changes, keeping modifications minimal. Retained necessary methods, removed extra spaces, and ensured consistent spacing.

- Removed unnecessary spaces and empty lines.
- Retained @MJI methods as per feedback.
- Ensured consistent spacing in method signatures.

Please review the changes.

cyrille-artho commented:
Thanks for adding the code back. Please try to add it back in the same way it was before, as there are still two changes that merely reformat code, which is best avoided (an empty line at the beginning and reordered/reformatted methods at the end).

adysinghh commented:
@cyrille-artho I had made the changes please review.

cyrille-artho reviewed (changes requested):
Hi,
Please try to isolate the first change. If you are not familiar with "diff" and "patch", you can try to apply the first change again manually on a new branch and make a new PR.
Thank you for your perseverance!

shreyasCs012 commented:
the native peers method present in the `\src\peers\gov\nasa\jpf\vm` had a  "mangled" which was old so it has been corrected to the new location which is `\jdk\internal\misc`

cyrille-artho commented:
Subsumed by PR #492
